### PR TITLE
Add a C test library for easier testing of the generated bindings

### PR DESCRIPTION
**Pull request by cameronwhite**
This is related to #504 - I decided to experiment with this after writing a C library to test out #763, and found it really useful to have a simple C function (which I could also add printouts to, etc) to test against
I think this could make it easier to have more extensive tests for correctness, which can run without a display and without requiring a lot of boilerplate code to call a function with the parameter type we want to test (e.g. having to construct a GTK application and widgets)

Still a work in progress, but wanted to get some initial feedback on whether this would be worth adding?
Also a couple questions:
- Currently I have this as a separate script. I wasn't sure about making this part of the default build since it adds the requirement of having a C toolchain, gir-related tools, etc?
- The library is currently generated into the `Libs` folder like everything else, but it could perhaps go elsewhere. I'm not sure if there are any things to watch out for with this, e.g. would it be pulled into the Nuget packages?

TODOs
- [x] Needs testing on other platforms, currently only tested on macOS
- [x] Add to CI build script

### CODE REVIEW

**Comment by badcel** [comment]
I did not look at the code but the idea is very welcome!

See #504

**Comment by cameronwhite** [comment]
I've got the Windows build sorted out now, so this is building and running the test successfully on all platforms 🎉

**Comment by badcel** [comment]
Can you update the reademe's Build section and include an `dotnet build Gir.Core.Libraries.slnf`?

Additionally I think it would be helpful to explain why the solution filter exists.

**Review by badcel — CHANGES_REQUESTED** [review]
Thanks for your work, very well done. This helps a lot to establish a much improved testability of the generator it is very much appreciated.

Please fix the remaining remarks and it is good to go.

File: src/Native/GirTestLib/primitivevaluetype.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include "primitivevaluetype.h"
```
Comment (badcel) [inline src/Native/GirTestLib/primitivevaluetype.c:1]: Should the folder: `src/Native/GirTestLib/` be part of the solution to allow easy editing?

**Comment by cameronwhite** [comment]
Thanks! I've pushed fixes for those issues

**Pull request merged**